testutil/server.go

```diff
@@ -173,26 +173,26 @@ type TestServer struct {
 
 // NewTestServer is an easy helper method to create a new Consul
 // test server with the most basic configuration.
-func NewTestServer() (*TestServer, error) {
-	return NewTestServerConfig(nil)
+func NewTestServer(name string) (*TestServer, error) {
```

Comment:
This is used by quite a few other projects so it's probably worth keeping these interfaces the same. How about making these call an underlying helper that takes the directory and making new variants that take `testing.T` and use the new `testutil.TempDir` under the hood?